modules/library/main/src/main/java/org/geotools/filter/function/DateDifferenceFunction.java

```diff
@@ -51,11 +66,26 @@ public int getArgCount() {
     public Object evaluate(Object feature) {
         Date a = getExpression(0).evaluate(feature, Date.class);
         Date b = getExpression(1).evaluate(feature, Date.class);
-
         if (a == null || b == null) {
             return null;
         }
+        TimeUnit timeUnit = TimeUnit.MILLISECONDS;
+        if (params.size() == 3) {
+            try {
+                timeUnit = getExpression(2).evaluate(null, TimeUnit.class);
+                if (timeUnit == null) {
+                    throw new IllegalArgumentException(
+                            "The specified timeUnit should be one of "
+                                    + SUPPORTED_TIME_UNITS_STRING);
+                }
+            } catch (Exception e) {
+                throw new IllegalArgumentException(
+                        "Unable to parse the specified timeUnit which should be one of "
+                                + SUPPORTED_TIME_UNITS_STRING,
+                        e);
+            }
+        }
```

Comment:
This could be factored out in its own method, ``getTimeUnit(getExpression(2))``

Reply:
Done